jar-infer/src/main/java/com/uber/nullaway/jarinfer/DefinitelyDerefedParamsDriver.java

```diff
@@ -30,70 +34,201 @@
 import com.ibm.wala.ssa.ISSABasicBlock;
 import com.ibm.wala.ssa.SSAInstruction;
 import com.ibm.wala.types.ClassLoaderReference;
-import com.ibm.wala.types.MethodReference;
+import com.ibm.wala.util.collections.Iterator2Iterable;
 import com.ibm.wala.util.config.AnalysisScopeReader;
-import com.ibm.wala.util.strings.Atom;
-import com.ibm.wala.util.strings.ImmutableByteArray;
-import com.ibm.wala.util.strings.UTF8Convert;
 import com.ibm.wala.util.warnings.Warnings;
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
 import java.io.IOException;
 import java.util.*;
 
 /*
  * Driver for running {@link DefinitelyDerefedParams}
  */
 public class DefinitelyDerefedParamsDriver {
+  private static String aStubXPath =
+      "/Users/subarno/src/NullAway/jar-infer/build/reports/tests/test.astubx";
   /*
-   * Usage: DefinitelyDerefedParamsDriver ( class_file _dir, class_name, method_name, method_signature)
-   *
+   * Usage: DefinitelyDerefedParamsDriver ( path, package_name, [jar_flag])
+   * path: jar file OR directory containing class files
    * @throws IOException
    * @throws ClassHierarchyException
    * @throws IllegalArgumentException
    */
-  public static Set<String> run(
-      String classFileDir, String mainClass, String targetMethod, String targetMethodSignature)
+  public static HashMap<String, Set<Integer>> run(String path, String pkgName, boolean isJar)
+      throws IOException, ClassHierarchyException, IllegalArgumentException {
+    if (isJar) {
+      // Extract jar contents
+      // link: https://www.javaworld.com/article/2077548
+      Preconditions.checkArgument(path.endsWith(".jar"), "invalid jar path!");
+      String jarDir = path.substring(0, path.lastIndexOf('.'));
+      System.out.println("extracting " + path + "...");
+      java.util.jar.JarFile jar = new java.util.jar.JarFile(path);
+      java.util.Enumeration enumEntries = jar.entries();
```

Comment:
I'd prefer it if you just imported these classes, unless there is a clash somewhere (instead of using the fully qualified name everywhere).